### PR TITLE
33144 checkbox radio button spacing

### PR DESCRIPTION
**Pull request by pdavies88**
## Description
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/33114

## Testing done
Screen Shot 2021-12-07 at 9 06 28 AM

## Screenshots
Screen Shot 2021-12-07 at 9 07 09 AM
Screen Shot 2021-12-07 at 9 07 16 AM
Screen Shot 2021-12-07 at 9 07 23 AM
Screen Shot 2021-12-07 at 9 07 29 AM
Screen Shot 2021-12-07 at 9 07 41 AM

## Acceptance criteria
- [X] Design guidelines should be used to create a consistent look and feel between checkboxes and radio buttons

## Definition of done
- [ ] Documentation has been updated, if applicable
- [X] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Comment by pdavies88** [comment]
PR is dependent on #234 as the Checkbox Group is styled as a part of this ticket

File: packages/web-components/src/components/va-radio/va-radio.css
Comment (k80bowman) [inline packages/web-components/src/components/va-radio/va-radio.css:21]: Should we be using em or rem in these margins? (this one and the one below)

Comment (pdavies88) [reply packages/web-components/src/components/va-radio/va-radio.css:21]: @k80bowman @bkjohnson honestly I'm not sure, in some places we use `em` and others we use `rem` I mainly did `1em` for the spacing because that is what is on the p tags and I wanted to make sure the vertical spacing stayed consistent

Comment (andresriveratoro) [reply packages/web-components/src/components/va-radio/va-radio.css:21]: Thanks for bringing this up. Up until now I've been using the existing styles from React components for the most part. I think we should decide on one and prefer `rem` because it's consistent and predictable.

Comment (k80bowman) [reply packages/web-components/src/components/va-radio/va-radio.css:21]: I agree with @andresriveratoro, we should prefer `rem` for things like padding and margins. I wanted to ask before recommending that, though, and make sure there wasn't a specific reason were using `em` here.

Comment (pdavies88) [reply packages/web-components/src/components/va-radio/va-radio.css:26]: Sounds good switching to `rem`

File: packages/web-components/src/components/va-radio-option/va-radio-option.css
Comment (pdavies88) [reply packages/web-components/src/components/va-radio-option/va-radio-option.css]: @andresriveratoro good catch, just updated and pushed

**Review by bkjohnson — COMMENTED** [review]
Are the `<va-checkbox-group>` changes supposed to be in here?

**Comment by pdavies88** [comment]
> PR is dependent on #234 as the Checkbox Group is styled as a part of this ticket

@bkjohnson this is the styling portion of the ticket so I needed Checkbox Group to be present so I could make sure that the Checkbox Group and Radio Button Group Styling matches for this ticket

**Comment by GnatalieH** [comment]
@pdavies88 what spacing and alignment specs are you using for this work? I'm wondering if we're going to implement the consistently spaced and aligned checkbox and radio button groups that I made designs for a while back. Thanks.

**Comment by pdavies88** [comment]
@GnatalieH Using the Sketch from here: https://www.sketch.com/s/cbfcac05-b6fa-4f3d-a1b6-5defe1dd79c6/a/JnmM5mb
The deployed version can be seen here: 
Checkbox Group: https://60f9b557105290003b387cd5-yywgwnqggb.chromatic.com/?path=/docs/components-va-checkbox-group--default
Radio Buttons: https://60f9b557105290003b387cd5-yywgwnqggb.chromatic.com/?path=/docs/components-va-radio--default
If you'd like me to update anything you see please feel free to comment and I can adjust

**Comment by GnatalieH** [comment]
@pdavies88 looks fab, thanks!

**Review by bkjohnson — COMMENTED** [review]
I poked around in storybook and the styling changes look ok to me, but I'd rather let Katy or Natalie give the final say on this.

---

Also I just noticed this - we probably want to use backticks for this:

image

https://github.com/department-of-veterans-affairs/component-library/blob/c5c85d2045f90b5fb048607f37340e7985273f3b/packages/web-components/src/components/va-checkbox/va-checkbox.tsx#L55

**Comment by pdavies88** [comment]
@bkjohnson I updated the `<input>` issue should be rendering properly now also Natalie said the PR looked good in the comments above. I'll wait for Katy's review on Monday but let me know if you see anything else to update

**Review by k80bowman — APPROVED** [review]
I'm going to go ahead and approve because this PR looks good to me.

However, I found another problem in the Story for this component and, if it's a quick fix, I'd appreciate it if we could just take care of it. If it's not a quick fix, I'm happy to create another ticket for it.

Would you happen to know why the Story for this checkbox is saying the error prop is looking for an object?
image

**Comment by pdavies88** [comment]
> I'm going to go ahead and approve because this PR looks good to me.
> 
> However, I found another problem in the Story for this component and, if it's a quick fix, I'd appreciate it if we could just take care of it. If it's not a quick fix, I'm happy to create another ticket for it.
> 
> Would you happen to know why the Story for this checkbox is saying the error prop is looking for an object? image

@k80bowman I adjusted with the fix in the most recent commit. Issue was that the Error prop was looking for `string | HTMLElement` instead of just `string`. It has been fixed with the most recent push
Screen Shot 2022-01-24 at 9 59 36 AM
Screen Shot 2022-01-24 at 9 59 45 AM

**Pull request merged**
